### ISSUE TITLE
BRAM Reset API

### ISSUE THREAD

**Issue by dkoeplin** [stanford-ppl/spatial#32]
Should BRAM reset be supported as a user-facing API? Is this intuitive enough to include in Spatial?

**Comment by mattfel1** [comment]
I don't think it is necessary.   It is simple enough to explicitly flush a memory.

Although with GEMM, it would be really convenient to have a tile transfer from bigger brams to smaller brams with the same API as offchip tile loaders.  So if we decide to allow multiple levels of hierarchy for on-chip memories and support a tile transfer API, maybe we can also use the same API to specify a flush as well.
